File: AEDS2/Aulai/estudos/pilha.java
  
 



 class celula{
    public int elemento;
    public celula prox;
    public celula(){
        this(0);
    }
    public celula(int x){
        this.elemento=x;
        this.prox=null;
    }
}

    public class pilha{
        public celula topo;
        public pilha(){
            topo=null;
        }
        
        public void inserir(int x){
            celula tmp= new celula(x);
            tmp.prox=topo;
            topo=tmp;
            tmp=null;
        }


        public int remover(){
            celula tmp=topo;
            int elemento=tmp.elemento;
            topo=topo.prox;
            tmp.prox=null;
            tmp=null;
            return elemento;
        }


        public void mostrar(){
            for (celula i=topo; i!=null; i=i.prox){
            System.out.println(" "+i.elemento);
            }
        }


            // celula maiorpilha(){
            //     celula retorno= null;
            //     int contador, maior=0;
            //     for(celula i=this.topo; i!=null; i=i.prox){
            //         contador=0;
            //         for(celula j=i.topo; j!=null; j=j.prox){
            //             contador++;
            //         }
            //     }
            // }

        
        
    }

File: AEDS2/Aulai/pilha.c
#include <stdio.h>
#include <stdlib.h>

typedef struct  celula
{
    int elemento;
    celula *prox;

}celula;
 
 celula *novacelula(int elemento){
    celula *nova = (celula *)malloc(sizeof (celula));
    nova->elemento= elemento;
    nova->prox= NULL;
    return nova;
 }

 typedef struct pilha{
    celula *topo;

    void start(){
        topo=NULL;
    }
    void inserir(pilha *pilhao, int x){
        celula *tmp= novacelula(x);
        tmp-> prox= pilhao->topo;
        pilhao->topo=tmp;
        tmp=NULL;
    
    }

    int remover(int x){
        celula *tmp=topo;
        int elemento= topo->elemento;
        topo=topo->prox;
        tmp->prox=NULL;
        free(tmp);
        tmp=NULL;
        return elemento;
    }
    void mostrar(){
        celula *i;
        for(*i=topo; i!=NULL; i=i->prox){
            printf("%d", i->elemento);
        }
    }




 }pilha;
